binlog_streamer.go

```diff
@@ -169,6 +177,8 @@ func (s *BinlogStreamer) Run() {
 		default:
 			s.updateLastStreamedPosAndTime(ev)
 		}
+
+		s.StateTracker.UpdateLastStreamedBinlogPosition(s.lastStreamedBinlogPosition)
```

Comment:
putting this call here is confusing because of the access to `s.lastStreamedBinlogPosition` which is updated somewhere completely different. why not put it in the `updateLastStreamedPosAndTime` method?

also, why don't we want to update state when we hit a `RotateEvent`?